go/types/map.go

```diff
@@ -227,6 +227,18 @@ func (m Map) IterAll(cb mapIterAllCallback) {
 	})
 }
 
+func (m Map) IterRange(start, end Value, cb mapIterAllCallback) {
```

Comment:
Did you see my comments in the previous patch?

It was basically that IMO this should be `IterFrom(start Value, cb mapIterCallback)` so that:
- caller can decide boundary conditions.
- caller can bail early.
it's easy enough for call site to go from
```
map.IterRange(start, end, func(k, v Value) {
  ...
});
```
to
```
map.IterFrom(start, func(k, v Value) (stop bool) {
  ...
  return end.Less(v); // or !v.Less(end)
})
```